com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -440,9 +464,9 @@ internal void InternalOnNetworkSpawn()
 
         internal void InternalOnNetworkDespawn()
         {
-            IsSpawned = false;
             UpdateNetworkProperties();
             OnNetworkDespawn();
+            IsSpawned = false;
```

Comment:
This is tricky. Not strictly needed. But as I explored I noticed the inconsistency. If `IsSpawned` is set before `OnNetworkSpawn`, it should be reset _after_  `OnNetworkDespawn` for consistency.

Reply:
The idea here is that when you are "despawning" you should not be considered "spawned".  Really, IsSpawned is meant to be used in places where you want an Update or FixedUpdate to "exit early" until the associated NetworkObject is spawned.  Due to the frequency in which it would be called under these conditions, it became an issue with performance and would require one to do more than one check (i.e. typically something like `NetworkManager.Singleton.IsListening && IsSpawned`) which now can be done just by checking IsSpawned.